classes/CartRule.php

```diff
@@ -408,9 +408,7 @@ public static function getCustomerCartRules(
         $free_shipping_only = false,
         $highlight_only = false
     ) {
-        if (!CartRule::isFeatureActive()
-            || !CartRule::haveCartRuleToday($id_customer)
-        ) {
+        if ($active && (!CartRule::isFeatureActive() || !CartRule::haveCartRuleToday($id_customer))) {
```

Comment:
So, if we don't want vouchers only and CartRule is disabled, we run everything? :thinking:

Reply:
thing is that if we have only one voucher (as described in Issue), and disable it, `CartRule` feature is treated as non active, so we need to omit this check

I was thinking about adding a new param like `$forceDisplay = false` but not sure if that's a good idea

Reply:
I'm also not sure it's a good idea to display things if CartRule feature is not active :thinking:

Reply:
It's tricky because we shouldn't care if the feature is active or not, we need to display all customer vouchers regardless of feature activity, especially if feature state is not related to our decision but to... state and number of vouchers

Reply:
It's weird :thinking: So the CartRule::isFeatureActive should be useless. Maybe we have to ping @PrestaShop/product-team and  @Robin-Fischer-PS

Reply:
We also can't retrieve cart rules that are not active :thinking:

Reply:
It's not that useless because this check is helpful in front-office

> We also can't retrieve cart rules that are not active 🤔

yes, and that's the point, we need to retrieve whatever we want to solve this issue, I could separate this query and create a dedicated method just to get what we need w/o checks, and left checks inside this old method, but naming  this new one would be challenging :D

Reply:
Maybe it's a better solution, having a method for one usecase, and not one method for all usecases :p